suite/tests/linux/signal_pre_syscall.c

```diff
@@ -62,59 +58,120 @@ fail(const char *s)
     exit(1);
 }
 
+/* The signal handler copies one global to the other so that we can
+ * discover when the signal happened.
+ */
 void
-handler(int signum, siginfo_t *info, void *ucontext_)
+handler(int signum, siginfo_t *info, void *ucontext)
 {
-    /* We just need the handler to exist; it needn't do anything but sigreturn. */
+    flag2 = flag1;
 }
 
-void
-setup(void)
+/* Return -1 if the signal happened before the critical block, 0 if it
+ * happened during the critical block, or 1 if it happened after the
+ * critical block or the timer was cancelled. Under DynamoRIO the
+ * signal will never appear to have happened during the critical block.
```

Comment:
Maybe add to the comment that this is because we delay async signals to the end of the block